core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -589,10 +589,11 @@ private[spark] class BlockManager(
   private def doGetRemote(blockId: BlockId, asBlockResult: Boolean): Option[Any] = {
     require(blockId != null, "BlockId is null")
     val locations = Random.shuffle(master.getLocations(blockId))
+    val timeout = conf.getInt("spark.storage.fetchBlockTimeout", 1000) // seconds
```

Comment:
broadcast timeout should be different from spark.network.timeout. netowkr.timeout is only for establishing connection or setting a small data. broadcast timeout can be quite large (e.g. reading 2g of data).